packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1345 @@
+                ) {
+                    return this.scrollToAndFocusRow(rowIndex, scrollOptions);
+                }
+                return false;
+            }
+            default:
+                break;
+        }
+
+        return false;
+    }
+
+    private updateFocusStateFromActiveElement(setRowFocus: boolean): void {
+        // If the user is interacting with the table with non-keyboard methods (like mouse), we need to
+        // update our focus state based on the current active/focused element
+        const activeElement = this.getActiveElement();
+        if (activeElement) {
+            const row = this.getContainingRow(activeElement);
+            if (row) {
+                if (!(row instanceof TableGroupRow)) {
+                    const cell = this.getContainingCell(activeElement);
+                    if (cell) {
+                        const columnIndex = this.table.visibleColumns.indexOf(
+                            cell.column!
+                        );
+                        if (cell.actionMenuButton === activeElement) {
+                            this.setCellActionMenuFocusState(
+                                row.rowStateIndex!,
+                                columnIndex,
+                                false
+                            );
+                            return;
+                        }
+                        const contentIndex = cell.cellView.tabbableChildren.indexOf(
+                            activeElement
+                        );
+                        if (contentIndex > -1) {
+                            this.setCellContentFocusState(
+                                contentIndex,
+                                row.rowStateIndex!,
+                                columnIndex,
+                                false
+                            );
+                            return;
+                        }
+                    }
+                }
+                if (
+                    setRowFocus
+                    && this.hasRowOrCellFocusType()
+                    && this.rowIndex !== row.rowStateIndex
+                ) {
+                    this.setRowFocusState(row.rowStateIndex);
+                }
+            }
+        }
+    }
+
+    private focusElement(
+        element: HTMLElement,
+        focusOptions?: FocusOptions
+    ): void {
+        const previousActiveElement = this.getActiveElement();
+        if (previousActiveElement !== element) {
+            this.setElementFocusable(element, true);
+            this.isCurrentlyFocusingElement = true;
+            element.focus(focusOptions);
+            this.isCurrentlyFocusingElement = false;
+            if (
+                previousActiveElement
+                && this.isInTable(previousActiveElement)
+            ) {
+                this.setElementFocusable(previousActiveElement, false);
+            }
+        }
+    }
+
+    private setElementFocusable(
```

Comment:
Is there a strong guaruntee this is a menu button that is passed in is only the action menu button and not part of a column's cell view? Can you include a comment for how that's guarunteed?
There are several code paths that end up here so not sure how it's guarunteed, maybe need some kind of check?

Reply:
You're right, this function was assuming that menu button implied cell action menu button, which may not be true in the future.

Since the `setActionMenuButtonFocused(button, false)` code path was already covered by `onCellActionMenuBlur()`, moved the `true` code path to be handled in `focusRowElement()`. So the menu button special case is gone from this function now (it just sets `tabIndex`).